packages/@react-aria/utils/src/runAfterTransition.ts

```diff
@@ -91,9 +91,23 @@ if (typeof document !== 'undefined') {
   }
 }
 
+/**
+ * Cleans up any elements that are no longer in the document.
+ * This is necessary because we can't rely on transitionend events to fire
+ * for elements that are removed from the document while transitioning.
+ */
+function cleanupDetachedElements() {
+  for (const [element] of transitionsByElement) {
+    if (element instanceof HTMLElement && !element.isConnected) {
```

Comment:
```suggestion
    let windowObject = getOwnerWindow(element);
    if (element instanceof windowObject.HTMLElement && !element.isConnected) {
```

Reply:
I'm fine with just checking the `isConnected` seems like it would handle all the cases needed

Reply:
Updated.